src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocsummarylocation/InputJavadocSummaryLocationTrailingSpace.java

```diff
@@ -0,0 +1,51 @@
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocsummarylocation;
+
+/**
+ * config:  default
+ * location="SECOND_LINE"
+ */
+public final class InputJavadocSummaryLocationTrailingSpace {
+
+    /**  
+     * ^^ There is a trailing space in the first line. // OK
+     */
+    private int space;
+
+    /**********************
+     * Trailing asterisks in the first line * // OK
+     *
+     **********************/
+    private int asterisk;
+
+    /**********************              
+     * Trailing asterisks, then spaces ^^ in the first line * // OK
+     *
+     **********************/
+    private int asteriskSpace;
+
+    /** ******
+     * There is a space, then leading asterisks in the first line. // OK
+     */
+    private int spaceAsterisk;
+
+    /**      ***************************************            
+     * There is a spaces, then leading asterisks, then spaces ^^ in the first line. // OK
+     */
+    private int spaceAsteriskSpace;
+
+    /*** **
+     * There is a leading asterisk, then a space, then asterisks in the first line.
```

Comment:
please convert this class to have public methods.
I did on my local and generated javadocs on jdk8:
![image](https://user-images.githubusercontent.com/812984/66696151-4a69e880-ec7e-11e9-87af-8a0279f10d41.png)

as you really wanted to work as javadoc tool ... this case is not problematoic for javadoc ....
so no violation is expected.

When we started discussion on such cases, I recommended to consider all chracters after `/**` as summary begin. If you do not want go such way .... lets declare that this Check plan to follow javadoc tool.

Reply:
Done. Test cases converted to methods. Added note for the corner case to the xdocs. 

To mimic the javadoc tool's behavior, we need to make changes to the parser.

@romani @rnveach I'll start a new issue if you both agree with this proposal:

Add new javadoc tag `ADDITIONAL_ASTERISK` for the `*` symbols right after `LEADING_ASTERISK` and `COMMENT_START`

For this comment
```
123456789
/****  **  
```
1,2,3 - comment start
4,5 - additional asterisks (ignored by javadoc tool)
6,7 - whitespace before leading asterisk
8 - leading asterisk
9 - additional asterisks (ignored by javadoc tool)

Since this new token should be ignored during processing, there should be no compatibility problems. Some `TEXT` tags will change from something like `* the text` to just `the text` prepended with `ADDITIONAL_ASTERISK` and `WS`.

Reply:
I originally was against mimicking javadoc.  So as you do not want to be strict, we need to ignore the spaces and asterisks. Test input shows that Check behavior is weird.

Is it made side effects to extend your regexp with asterisk symbol ?

Outbound issue is ok, but it most likely go to to tail of backlog. I would rather spend time to let checkstyle parse java sources property (jdk13 multiline strings, is too long awaited feature in java, users would for sure sacrifice checkstyle to start using this feature of language).